compiler/src/dotty/tools/dotc/interactive/SourceTree.scala

```diff
@@ -21,7 +21,11 @@ case class SourceTree(tree: tpd.NameTree, source: SourceFile) {
     if (treePos.isZeroExtent || tree.name.toTermName == nme.ERROR)
       NoSourcePosition
     else {
-      val nameLength = tree.name.stripModuleClassSuffix.show.toString.length
+      // Constructors are named `<init>` in the trees, but `this` in the source.
+      val nameLength = tree.name.stripModuleClassSuffix match {
```

Comment:
`<init>` never comes with a module class suffix. So the `stripModuleClassSuffix` could go in the `other` branch.